File: day01.py
with open("inputs/day01.txt", "r") as file:
    data = file.read()

lines = data.strip().split("\n")

instructions = [(line[0], int(line[1:])) for line in lines]


currentValue = 50

part1Counter = 0
part2Counter = 0

for instruction in instructions:
    direction, distance = instruction

    dirValue = 1 if direction == "R" else -1

    for i in range(distance):
        currentValue += dirValue
        currentValue %= 100

        if currentValue == 0:
            part2Counter += 1

    if currentValue == 0:
        part1Counter += 1


print(currentValue)
print(part1Counter)
print(part2Counter)

File: day01.js
import { readFileSync } from "fs";

const input = readFileSync(`inputs/day01.txt`, "utf8");

const data = input
  .trim()
  .split("\n")
  .map((line) => [line.trim()[0], parseInt(line.trim().slice(1))]);

function part1() {
  let currentValue = 50;
  let zeroCount = 0;

  for (const [direction, distance] of data) {
    if (direction === "L") {
      currentValue -= distance;
    } else if (direction === "R") {
      currentValue += distance;
    } else {
      throw new Error(`Invalid direction: ${direction}`);
    }

    currentValue %= 100;

    if (currentValue < 0) {
      currentValue += 100;
    }

    if (currentValue === 0) {
      zeroCount++;
    }
  }

  console.log(currentValue);
  console.log(zeroCount);
}

part1();

function part2() {
  let currentValue = 50;
  let zeroCount = 0;

  for (const [direction, distance] of data) {
    const delta = direction === "L" ? -1 : 1;

    for (let step = 0; step < distance; step++) {
      currentValue += delta;
      currentValue %= 100;

      if (currentValue < 0) {
        currentValue += 100;
      }

      if (currentValue === 0) {
        zeroCount++;
      }
    }
  }
  console.log(currentValue);
  console.log(zeroCount);
}

part2();
